meteor/server/api/ingest/packageInfo.ts

```diff
@@ -0,0 +1,161 @@
+	// 			200
+	// 		)
+	// 	})
+	// }
+}
+
+export function onUpdatedPackageInfo(packageId: ExpectedPackageId, doc: PackageInfoDB | null) {
+	logger.info(`PackageInfo updated "${packageId}"`)
+
+	const pkg = ExpectedPackages.findOne(packageId)
+	if (!pkg) {
+		logger.error(`onUpdatedPackageInfo: Received update for missing package: "${packageId}"`)
+		return
+	}
+
+	if (pkg.listenToPackageInfoUpdates) {
+		switch (pkg.fromPieceType) {
+			case ExpectedPackageDBType.PIECE:
+			case ExpectedPackageDBType.ADLIB_PIECE:
+			case ExpectedPackageDBType.ADLIB_ACTION:
+			case ExpectedPackageDBType.BASELINE_ADLIB_PIECE:
+			case ExpectedPackageDBType.BASELINE_ADLIB_ACTION:
+			case ExpectedPackageDBType.RUNDOWN_BASELINE_OBJECTS: {
+				const existingEntry = pendingPackageUpdates.get(pkg.rundownId)
+				if (existingEntry) {
+					// already queued, add to the batch
+					existingEntry.push(pkg._id)
+				} else {
+					pendingPackageUpdates.set(pkg.rundownId, [pkg._id])
+				}
+
+				lazyIgnore(
+					`onUpdatedPackageInfoForRundown_${pkg.rundownId}`,
+					() => {
+						const packageIds = pendingPackageUpdates.get(pkg.rundownId)
+						if (packageIds) {
+							pendingPackageUpdates.delete(pkg.rundownId)
+							onUpdatedPackageInfoForRundown(pkg.rundownId, packageIds)
+						}
+					},
+					1000
+				)
+				break
+			}
+			case ExpectedPackageDBType.BUCKET_ADLIB:
+			case ExpectedPackageDBType.BUCKET_ADLIB_ACTION:
+			case ExpectedPackageDBType.STUDIO_BASELINE_OBJECTS:
+				// Ignore, as we can't handle that for now
+				break
+			default:
+				assertNever(pkg)
+				break
+		}
+	}
+}
+
+const pendingPackageUpdates = new Map<RundownId, Array<ExpectedPackageId>>()
+
+function onUpdatedPackageInfoForRundown(rundownId: RundownId, packageIds: Array<ExpectedPackageId>) {
+	if (packageIds.length === 0) {
+		return
+	}
+
+	const tmpRundown = Rundowns.findOne(rundownId)
+	if (!tmpRundown) {
+		logger.error(
+			`onUpdatedPackageInfoForRundown: Missing rundown "${rundownId}" for packages "${packageIds.join(', ')}"`
+		)
+		return
+	}
+
+	return runIngestOperationWithCache(
+		'onUpdatedPackageInfoForRundown',
+		tmpRundown.studioId,
+		tmpRundown.externalId,
+		(ingestRundown) => {
+			if (!ingestRundown)
+				throw new Meteor.Error('onUpdatedPackageInfoForRundown called but ingestData is undefined')
+			return ingestRundown // don't mutate any ingest data
+		},
+		async (cache, ingestRundown) => {
+			if (!ingestRundown)
+				throw new Meteor.Error('onUpdatedPackageInfoForRundown called but ingestData is undefined')
+
+			/** All segments that need updating */
+			const segmentsToUpdate = new Set<SegmentId>()
+
+			for (const packageId of packageIds) {
+				const pkg = cache.ExpectedPackages.findOne(packageId)
+				if (pkg) {
+					if (
+						pkg.fromPieceType === ExpectedPackageDBType.PIECE ||
+						pkg.fromPieceType === ExpectedPackageDBType.ADLIB_PIECE
+					) {
+						segmentsToUpdate.add(pkg.segmentId)
+					}
+					// TODO - trigger baseline regenerateion
```

Comment:
Does this TODO neded to be done? Sounds important?

Reply:
We're leaving a stub for now, to move forward. will implement properly later.